src/jaxsim/api/data.py

```diff
@@ -43,6 +43,24 @@ class JaxSimModelData(common.ModelDataWithVelocityRepresentation):
         default_factory=lambda: jnp.array(0, dtype=jnp.uint64)
     )
 
+    def __hash__(self) -> int:
+
+        return hash(
+            (
+                hash(self.state),
+                hash(tuple(self.gravity.flatten().tolist())),
```

Comment:
`self.gravity` has shape `(3,)`, so probably the `.flatten()` is not necessary
```suggestion
                hash(tuple(self.gravity.tolist())),
```

Reply:
The operator `jnp.atleast1d(X).flatten()` is there mainly for making the hash function work on vectorized pytrees.